### ISSUE TITLE
predictions for demo notebook look bad

### ISSUE THREAD

**Issue by tfriedel** [instadeepai/InstaGeo-E2E-Geospatial-ML#32]
I ran the demo notebook on colab.

Besides just running it I took the mean / std values that were calculated in the first pass:
[192.44924926757812, 372.3095703125, 290.4117736816406, 1582.2117919921875, 348.9172058105469, -276.9428405761719]
[3528.393798828125, 3539.972412109375, 3493.9990234375, 3670.82666015625, 3236.271728515625, 2996.205322265625]

I then overwrote the mean / std values in locust.yaml with those and continued with the 5 epoch training run. 

For the checkpoint path I inserted the path to the last checkpoint (from epoch 4).

These are the results from the evaluation:
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃        Test metric        ┃       DataLoader 0        ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│     test_Acc_0_epoch      │    0.7029629945755005     │
│     test_Acc_1_epoch      │    0.4037036895751953     │
│     test_IoU_0_epoch      │    0.5187915563583374     │
│     test_IoU_1_epoch      │    0.24437761306762695    │
│  test_Precision_0_epoch   │    0.6805270314216614     │
│  test_Precision_1_epoch   │    0.44798535108566284    │
│    test_Recall_0_epoch    │    0.7029629945755005     │
│    test_Recall_1_epoch    │    0.4037036895751953     │
│      test_aAcc_epoch      │    0.5833333730697632     │
│      test_loss_epoch      │    0.7383369207382202     │
│      test_mIoU_epoch      │    0.3815845847129822     │
│    test_roc_auc_epoch     │    0.44695472717285156    │

Then I ran inference and looked at the resulting tiles:

![Image](https://github.com/user-attachments/assets/c0aef476-26bc-4c04-bfbc-a49007b09b71)

![Image](https://github.com/user-attachments/assets/cf65d3f9-c752-43f3-a350-c25fd5663324)

They look quite odd and repetitive. 
Is this expected? I know the training was done with a very limited small dataset, so I can't rule out that we get bad results from that.
However the metrics don't look so bad to suggest this.
Can you please check if that looks plausible or if I missed something?

**Comment by tfriedel** [comment]
Seems it is indeed the result of the short training. When I train for 20 epochs result look better. However there's some white pixels near the tiles' borders. Maybe the sliding window method needs some better handling of these areas?

<img width="527" alt="Image" src="https://github.com/user-attachments/assets/c99b1e38-49f2-412f-bd1c-4cf07346050d" />
